src/main.cpp

```diff
@@ -1947,7 +1947,7 @@ bool ReadBlockHeaderFromDisk(CBlock &block, const CDiskBlockPos &pos) {
 
 CAmount GetBlockSubsidy(int nHeight, const Consensus::Params &consensusParams, int nTime) {
     // Just want to make sure no one gets a dime before 28 Sep 2016 12:00 AM UTC
-    if (nTime < nStartRewardTime && !consensusParams.IsTestnet())
+    if (nTime < nStartRewardTime && consensusParams.IsMain())
```

Comment:
I don't think this is the right fix. Regtest should similar to mainnet as much as possible. We should try to fix in another way.

Reply:
This will be remove after we merge from bitcoin. we can leave it and comment testcase that mocktime as the time before constant start time.

Reply:
If Bitcoin already removed this check then we should remove it now instead of comment out test cases. @psolstice do you see any problems if we remove this check?